extensions/functions_arithmetic.yaml

```diff
@@ -1614,3 +1614,141 @@ window_functions:
         decomposable: NONE
         return: i64?
         window_type: PARTITION
+  - name: "first_value"
+    description: >
+      Returns the first value in the window.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "last_value"
+    description: >
+      Returns the last value in the window.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "nth_value"
+    description: >
+      Returns a value from the nth row based on the `window_offset`. `window_offset` should
+      be a positive integer. If the value of the `window_offset` is outside the range
+      of the window, `null` is returned.
+
+      The `on_domain_error` option governs behavior in cases where `window_offset` is not
+      a positive integer or `null`.
+    impls:
+      - args:
+          - value: any1
+            name: expression
+          - value: i32
+            name: window_offset
+        options:
+          on_domain_error:
+            values: [ NULL, ERROR ]
+        nullability: DECLARED_OUTPUT
+        decomposable: NONE
+        return: any1?
+        window_type: PARTITION
+  - name: "lead"
+    description: >
+      Return a value from a following row based on a specified physical offset.
+      This allows you to compare a value in the current row against a following row.
+
+      The `expression` is evaluated against a row that comes after the current row based
+      on the `window_offset`.  The `window_offset` should be a positive integer and is set to
+      1 if not specified explicitly. If the `window_offset` is negative, the expression
```

Comment:
This is my fault for not being clear but I think `window_offset` is incorrect for `lag` and `lead`.  In other words...

| Function | Variable Name |
| --- | --- |
| nth_value | window_offset |
| lag | row_offset |
| lead | row_offset |

This is because, in `nth_value` the offset is from the start of the window.  In `lag` and `lead` the offset is from the current row.

Reply:
updated!